pkg/collector/corechecks/agentprofiling/agentprofiling.go

```diff
@@ -0,0 +1,238 @@
+type Config struct {
+	MemoryThreshold string `yaml:"memory_threshold"`
+	CPUThreshold    int    `yaml:"cpu_threshold"`
+	TicketID        string `yaml:"ticket_id"`
+	UserEmail       string `yaml:"user_email"`
+}
+
+// Check is the check that generates a flare with profiles when the core agent's memory or CPU usage exceeds a certain threshold
+type Check struct {
+	core.CheckBase
+	instance        *Config
+	flareGenerated  bool
+	flareComponent  flare.Component
+	agentConfig     config.Component
+	lastCPUTimes    *cpu.TimesStat
+	lastCheckTime   time.Time
+	memoryThreshold uint // Parsed memory threshold in bytes
+}
+
+// Factory creates a new instance of the agentprofiling check
+func Factory(flareComponent flare.Component, agentConfig config.Component) option.Option[func() check.Check] {
+	return option.New(func() check.Check {
+		return newCheck(flareComponent, agentConfig)
+	})
+}
+
+// newCheck creates a new instance of the agentprofiling check
+func newCheck(flareComponent flare.Component, agentConfig config.Component) check.Check {
+	return &Check{
+		CheckBase:      core.NewCheckBase(CheckName),
+		instance:       &Config{},
+		flareComponent: flareComponent,
+		agentConfig:    agentConfig,
+	}
+}
+
+// Parse parses the configuration for the agentprofiling check
+func (c *Config) Parse(data []byte) error {
+	return yaml.Unmarshal(data, c)
+}
+
+// Configure configures the agentprofiling check
+func (m *Check) Configure(senderManager sender.SenderManager, _ uint64, config, initConfig integration.Data, source string) error {
+	err := m.CommonConfigure(senderManager, initConfig, config, source)
+	if err != nil {
+		return err
+	}
+
+	err = m.instance.Parse(config)
+	if err != nil {
+		return err
+	}
+
+	// Parse memory threshold
+	m.memoryThreshold = size.ParseSizeInBytes(m.instance.MemoryThreshold)
+	return nil
+}
+
+// calculateCPUPercentage calculates the CPU percentage since the last check
+func (m *Check) calculateCPUPercentage(currentTimes *cpu.TimesStat) float64 {
+	if m.lastCPUTimes == nil {
+		m.lastCPUTimes = currentTimes
+		m.lastCheckTime = time.Now()
+		return 0.0
+	}
+
+	// Calculate time elapsed since last check
+	elapsed := time.Since(m.lastCheckTime).Seconds()
+	if elapsed <= 0 {
+		return 0.0
+	}
+
+	// Calculate CPU time delta
+	userDelta := currentTimes.User - m.lastCPUTimes.User
+	systemDelta := currentTimes.System - m.lastCPUTimes.System
+	totalDelta := userDelta + systemDelta
+
+	// Calculate percentage
+	cpuPercent := (totalDelta / elapsed) * 100.0
+
+	// Update last values
+	m.lastCPUTimes = currentTimes
+	m.lastCheckTime = time.Now()
+
+	return cpuPercent
+}
+
+// Run executes the agent profiling check, generating a flare with profiles if thresholds are exceeded
+func (m *Check) Run() error {
+	// Don't run again if the flare has already been generated
+	if m.flareGenerated {
+		log.Debugf("Flare with profiles already attempted to generate, skipping further check runs.")
```

Comment:
Looking back on this, in debug mode, this would spam the log with 1 line per 15s. The log line 235 might be enough and we could remove this. WDYT ?